include/dlaf/communication/datatypes.h

```diff
@@ -10,22 +10,25 @@
 
 #pragma once
 
-#include "dlaf/mpi_header.h"
-
-#include <complex>
+#include "dlaf/communication/internal/datatypes.h"
 
 namespace dlaf {
 namespace comm {
 
-/// @brief mapper between language types and basic MPI_Datatype
-template <typename T>
-struct mpi_datatype {
-  static MPI_Datatype type;
-};
-
-/// helper for mapping also custom types
-template <typename T>
-struct mpi_datatype<const T> : public mpi_datatype<T> {};
-
```

Comment:
I think this shouldn't be removed. With the approach outlined below, this is still useful for removing `const`.